runtime/src/accounts_db.rs

```diff
@@ -15546,4 +15546,140 @@ pub mod tests {
         let expected_alive_roots = [active_root].into_iter().collect();
         assert_eq!(result, expected_alive_roots, "extra: {}", extra);
     }
+
+    #[test]
+    fn test_unref_accounts() {
+        let empty_vec = Vec::default();
+        // pass 0 calls unref_accounts
+        // pass 1 calls clean_dead_slots_from_accounts_index, which calls unref_accounts
+        for pass in 0..2 {
```

Comment:
this pattern is odd to me. It seems simpler to just call the 2 functions rather than having a loop. As is, the same if block is repeated several times

Reply:
the setup and test results are all re-used for both functions. Some parameters are by value. And calling either function has side effects in the accounts index. It is a feature that these tests test both functions at the same time. I can simplify the 'if' into its own helper function.

Reply:
see if you like this better.